dipy/reconst/sfm.py

```diff
@@ -458,21 +475,45 @@ def fit(self, data, mask=None):
         isotropic = self.isotropic(self.gtab).fit(data, mask)
         flat_params = np.zeros((data_in_mask.shape[0],
                                 self.design_matrix.shape[-1]))
+        del data_in_mask
+        gc.collect()
+
         isopredict = isotropic.predict()
         if mask is None:
             isopredict = np.reshape(isopredict, (-1, isopredict.shape[-1]))
         else:
             isopredict = isopredict[mask]
 
-        for vox, vox_data in enumerate(flat_S):
-            # In voxels in which S0 is 0, we just want to keep the
-            # parameters at all-zeros, and avoid nasty sklearn errors:
-            if not (np.any(~np.isfinite(vox_data)) or np.all(vox_data == 0)):
-                fit_it = vox_data - isopredict[vox]
-                with warnings.catch_warnings():
-                    warnings.simplefilter("ignore")
-                    flat_params[vox] = self.solver.fit(self.design_matrix,
-                                                       fit_it).coef_
+        if n_threads > 1:
+            from collections import OrderedDict
+            try:
```

Comment:
We should use https://github.com/dipy/dipy/blob/master/dipy/utils/optpkg.py#L18 instead of this construction. 

Check out the sklearn import at the top of this file.

Reply:
Isn't the Collections module is a built-in module that comes with a base installation of Python? For now I've just moved the import up to the top of the file.

Reply:
For the joblib import